### DISCUSSION TITLE
Minions vs tty menus

### DISCUSSION THREAD

**Discussion by gerd-moellmann** [tarsius/minions#42] [Ideas]
Hi Jonas, this is a feature request.

I'm using Minions on a tty here, and I'm wondering if you could extend
it a bit to make it more useful on ttys.

The problem I have is that the popup menus on ttys don't allow
scrolling the menu.

On a GUI (Cocoa in my case), Minions' minor mode menu pops up, and if
it is longer than what can be displayed, the popup menu has a line at
the bottom/top with which one can scroll the menu. The current tty
menu code doesn't have such a feature, so one's stuck with what fits on
the frame and the rest in inaccessible.

Would it perhaps be possible to add to Minions, if on a tty, that it
puts a "more..." menu item before the menu gets too long, and then
display the rest in another menu and so on?

(Yes, the tty menus should be fixed, and the best way would probably
be to replace the current code with something based on tty child
windows. If that lands, but anyway, I don't see anyone including
myself, doing anything in the forseaable future.)

Thanks for Minions! I like it a lot.

**Comment by tarsius** [comment]
Thanks for the suggestion. This is something that I probably won't work on any time soon. I might, once I start using emacs in a terminal again. To be honest, I hope that by the time I do that, this has been addressed for tty menus in general.

**Comment by gerd-moellmann** [comment]
Ok thanksSent from my iPhoneOn 13. Dec 2024, at 16:26, Jonas Bernoulli ***@***.***> wrote:﻿
Thanks for the suggestion. This is something that I probably won't work on any time soon. I might, once I start using emacs in a terminal again. To be honest, I hope that by the time I do that, this has been addressed for tty menus in general.

**Comment by tarsius** [comment]
I just tried with Emacs 31.0.91 and that allows moving to items that were initially not visible. I used the arrow keys to do that.

Sinc that works now, I don't intend to add any additional tty support.
